ReactWindows/ReactNative/UIManager/UIViewOperationQueueCollection.cs

```diff
@@ -0,0 +1,606 @@
+/// <summary>
+/// NOTE: This file will be renamed to UIViewOperationsQueue.cs after code review.
+/// </summary>
+
+using Newtonsoft.Json.Linq;
+using ReactNative.Bridge;
+using ReactNative.Modules.Core;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Windows.ApplicationModel.Core;
+using Windows.UI.Core;
+
+namespace ReactNative.UIManager
+{
+    /// <summary>
+    /// This class acts as a buffer for command executed on
+    /// <see cref="NativeViewHierarchyManager"/>. It exposes similar methods as
+    /// mentioned classes but instead of executing commands immediately, it 
+    /// enqueues those operations in a queue that is then flushed from 
+    /// <see cref="UIManagerModule"/> once a JavaScript batch of UI operations
+    /// is finished.
+    /// There is one queue instance created for each top level window since
+    /// each of these windows run on a different thread.
+    /// </summary>
+    public class UIViewOperationQueue
+    {
+        private readonly ReactContext _reactContext;
+        private readonly ViewManagerRegistry _viewManagerRegistry;
+
+        private class QueueInstanceInfo
+        {
+            public UIViewOperationQueueInstance queueInstance;
+            public int rootViewCount;
+        }
+
+        private UIViewOperationQueueInstance _mainUiViewOperationsQueueInstance;
+
+        private readonly IDictionary<int, UIViewOperationQueueInstance> _reactTagToOperationQueue = new Dictionary<int, UIViewOperationQueueInstance>();
+        private readonly IDictionary<CoreDispatcher, QueueInstanceInfo> _dispatcherToOperationQueueInfo = new Dictionary<CoreDispatcher, QueueInstanceInfo>();
+
+        private bool _active;
+
+        /// <summary>
+        /// Instantiates the <see cref="UIViewOperationQueue"/>.
+        /// </summary>
+        /// <param name="reactContext">The React context.</param>
+        /// The native view hierarchy manager.
+        /// </param>
+        public UIViewOperationQueue(ReactContext reactContext, ViewManagerRegistry viewManagerRegistry)
+        {
+            _reactContext = reactContext;
+            _viewManagerRegistry = viewManagerRegistry;
+            _active = false;
+        }
+
+        /// <summary>
+        /// The native view hierarchy manager.
+        /// </summary>
+        internal UIViewOperationQueueInstance MainUIViewOperationQueue
```

Comment:
in UWP we found it more convenient for getters to use the short form eg MainUIViewOperationQueue => _main.....